gui-daemon/xside.c

```diff
@@ -1329,22 +1349,22 @@ static void update_wm_user_time(Ghandles *const g, const Window window,
             1);
 }
 
-/* handle local Xserver event: XKeyEvent
+/* handle local XInput event
  * send it to relevant window in VM
  */
-static void process_xevent_keypress(Ghandles * g, const XKeyEvent * ev)
+static void process_xievent_keypress(Ghandles * g, const XIDeviceEvent * ev)
 {
     struct msg_hdr hdr;
     struct msg_keypress k;
-    CHECK_NONMANAGED_WINDOW(g, ev->window);
-    update_wm_user_time(g, ev->window, ev->time);
+    CHECK_NONMANAGED_WINDOW(g, ev->event);
+    update_wm_user_time(g, ev->event, ev->time);
```

Comment:
```suggestion
    // yes, ev->event is the window number
    update_wm_user_time(g, ev->event, ev->time);
    if (ev->flags & XIKeyRepeat)
        return; // don't send key repeat events    
```

It’s probably best to call `update_wm_user_time` even if no event will be sent, since a key repeat should still prevent screen lock.  Otherwise the screen would lock too soon if a user held down a key for a long time.